process/transaction/shardProcess.go

```diff
@@ -482,22 +500,77 @@ func (txProc *txProcessor) processTxFee(
 		if err != nil {
 			return nil, nil, err
 		}
+
+		err = txProc.accounts.SaveAccount(acntSnd)
+		if err != nil {
+			return nil, nil, err
+		}
 	} else {
 		err := acntSnd.SubFromBalance(moveBalanceFee)
 		if err != nil {
 			return nil, nil, err
 		}
+
+		err = txProc.accounts.SaveAccount(acntSnd)
```

Comment:
you can move the saveAccount out from the if and else structure. Actually why do you need the saveAccount here in the processTxFee function ? You could do it outside of this, if processTxFee does not return error.